### DISCUSSION TITLE
Compare dateTime in where query

### DISCUSSION THREAD

**Discussion by xuala69** [simolus3/drift#2072] [Q&A]
Hello,

Is there a way to compare dateTime in a where query. Say I need to get all entries that are created after a specific dateTime, how would that be acheived?

Thank you

**Comment by simolus3** [comment]
Hi, you can use the methods defined on [this extension](https://drift.simonbinder.eu/api/drift/comparableexpr) to compare values in SQL, which includes date times.

> Say I need to get all entries that are created after a specific dateTime, how would that be acheived

Something like `where((row) => row.isBiggerOrEqualValue(yourMinDate))` should work.

**Reply by xuala69** [reply]
Thank you ... comparing with the value works great but what if I need to use a comparison that uses an Expression<DateTime>  as the factor... like  value.isBiggerOrEqual()

**Reply by simolus3** [reply]
> but what if I need to use a comparison that uses an Expression as the factor... like value.isBiggerOrEqual()

I'm not sure what you mean, `value.isBiggerOrEqual()` [exists](https://drift.simonbinder.eu/api/drift/comparableexpr/isbiggerorequal) and can be used to compare two expressions.

**Reply by xuala69** [reply]
I think my main question would be ... how do I convert/turn a dateTime variable into an Expression <DateTime> which is required to be passed as the parameter

**Reply by simolus3** [reply]
Ah, I see. You can just use `Variable<DateTime>(yourValue)` for that. It will generate a statement parameter in SQL. In fact, `isBiggerOrEqualValue(x)` just calls `isBiggerOrEqual(Variable(x))` internally.
